test/Benchmarks/Instrumentation/HttpInstrumentationBenchmark.cs

```diff
@@ -0,0 +1,87 @@
+﻿// <copyright file="HttpInstrumentationBenchmark.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using OpenTelemetry;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Tests;
+using OpenTelemetry.Trace;
+
+namespace Benchmarks.Instrumentation
+{
+    [MemoryDiagnoser]
+    public class HttpInstrumentationBenchmark
+    {
+        private const string SourceName = "http-client-test";
+        private const string ResourceName = "http-service-example";
+        private IDisposable serverLifeTime;
+        private TracerProvider openTelemetry;
+        private string url;
+        private HttpClient httpClient;
+        private ActivitySource source;
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+            this.serverLifeTime = TestHttpServer.RunServer(
+                (ctx) =>
+                {
+                    ctx.Response.StatusCode = 200;
+                    ctx.Response.OutputStream.Close();
+                },
+                out var host,
+                out var port);
+
+            this.openTelemetry = Sdk.CreateTracerProviderBuilder()
+                .AddHttpClientInstrumentation()
+                .SetResource(Resources.CreateServiceResource(ResourceName))
+                .AddSource(SourceName)
+                .Build();
+
+            this.url = $"http://{host}:{port}/";
+            this.httpClient = new HttpClient();
+            this.source = new ActivitySource(SourceName);
+        }
+
+        [GlobalCleanup]
+        public void GlobalCleanup()
+        {
+            this.httpClient.Dispose();
+            this.openTelemetry.Dispose();
+            this.serverLifeTime.Dispose();
+            this.source.Dispose();
+        }
+
+        [Benchmark]
+        public async Task InstrumentedHttpClient()
+        {
+            var httpResponse = await this.httpClient.GetAsync(this.url);
+            httpResponse.EnsureSuccessStatusCode();
+        }
+
+        [Benchmark]
+        public async Task InstrumentedHttpClientWithParentActivity()
+        {
+            using var parent = this.source.StartActivity("incoming request", ActivityKind.Server);
```

Comment:
can you move "incoming request" to const string.

Reply:
done.